### PR TITLE
Add new parsing apis

### PR DESCRIPTION
**Pull request by adrienball**
**Description**:
This PR introduces 2 new parsing APIs.
The first one allows to run intent classification only, and get back a list of intents along with their probability:

```python
>>> nlu_engine.get_intents("turn the lights on")
[
 {
   "intentName": "turnLightOn",
   "probability": 0.6363648460343694
 },
 {
   "intentName": null,
   "probability": 0.2580088944934134
 },
 {
   "intentName": "turnLightOff",
   "probability": 0.22791834836267366
 },
 {
   "intentName": "setTemperature",
   "probability": 0.181781583254962
 }
]
```

The second API allows to extract the slots when the intent is already known:

```python
>>> engine.get_slots(u"Hey, lights on in the lounge !", "turnLightOn")
[
 {
   "range": {
     "start": 22,
     "end": 28
   },
   "rawValue": "lounge",
   "value": {
     "kind": "Custom",
     "value": "living room"
   },
   "entity": "room",
   "slotName": "room"
 }
]
```

On top of these two new APIs, an optional `top_n` parameter is added to the `parse` method, allowing to perform intent parsing on the `top_n` most likely intents.

This should address https://github.com/snipsco/snips-nlu/issues/623, and cover the feature brought by @deeiip in https://github.com/snipsco/snips-nlu/pull/715.  

**Checklist**:
- [x] My PR is ready for code review
- [x] I have added some tests, if applicable, and run the whole test suite, including [linting tests](../linting_test.py)
- [x] I have updated the documentation, if applicable

### CODE REVIEW

File: snips_nlu/nlu_engine/nlu_engine.py [hunk 1]
```diff
@@ -114,95 +115,117 @@ def fit(self, dataset, force_retrain=True):
...
+                    continue
+                resolved_slots = self._resolve_slots(text, res[RES_SLOTS])
+                return parsing_result(text, intent=res[RES_INTENT],
+                                      slots=resolved_slots)
+            return empty_result(text)
+
+        intents_results = self.get_intents(text)
+        if intents is not None:
+            intents_results = [res for res in intents_results
+                               if res[RES_INTENT_NAME] in intents]
```
Comment (ClemDoum) [inline snips_nlu/nlu_engine/nlu_engine.py]:
```suggestion
                               if res[RES_INTENT_NAME] in set(intents)]
```

File: snips_nlu/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    """Raised when attempting to create a Snips NLU dataset using a wrong
+    format"""
+
+
+class EntityFormatError(SnipsNLUError):
+    """Raised when attempting to create a Snips NLU entity using a wrong
+    format"""
+
+
+class IntentFormatError(SnipsNLUError):
```
[the remark this answers is not in the document]
Comment (adrienball) [reply snips_nlu/exceptions.py]: Indeed

File: snips_nlu/intent_classifier/log_reg_classifier.py [hunk 3]
```diff
@@ -116,33 +116,49 @@ def get_intent(self, text, intents_filter=None):
             NotTrained: When the intent classifier is not fitted
 
         """
+        intents_results = self._get_intents(text, intents_filter)
+        if not intents_results or intents_results[0][RES_INTENT_NAME] is None:
+            return None
```
[the remark this answers is not in the document]
Comment (adrienball) [reply snips_nlu/intent_classifier/log_reg_classifier.py]: But then you don't pass the `intents_filter` parameter.

**Comment by timtutt** [comment]
When will there be a new release that includes this PR?

**Pull request merged**